File: String/Minimum Recolors to Get K Consecutive Black Blocks.go
func minimumRecolors(blocks string, k int) int {
    res := 101
    curr := 0
    start := 0

    for i, ch := range blocks {
        if ch == 'W' {
            curr += 1
        }
        if i - start + 1 > k {
            if blocks[start] == 'W' {
                curr -= 1
            }
            start += 1
        }
        if i - start + 1 == k {
            res = min(res, curr)
        }
    }
    return res
}

File: String/Minimum Recolors to Get K Consecutive Black Blocks.rs
impl Solution {
    pub fn minimum_recolors(blocks: String, k: i32) -> i32 {
        let block_ch: Vec<char> = blocks.chars().collect();
        let mut s = 0;
        let mut res = k;
        let mut curr = 0;

        for i in 0..blocks.len() {
            if block_ch[i] == 'W' {
                curr += 1;
            }
            if i - s + 1 > k as usize {
                if block_ch[s] == 'W' {
                    curr -= 1;
                }
                s += 1;
            }
            if i - s + 1 == k as usize {
                res = res.min(curr);
            }
        }
        res
    }
}